apps/dashboard/app/helpers/application_helper.rb

```diff
@@ -64,9 +68,20 @@ def app_icon_tag(app)
   def icon_tag(icon_uri)
     if %w(fa fas far fab fal).include?(icon_uri.scheme)
       fa_icon(icon_uri.host, fa_style: icon_uri.scheme)
-    else
+    elsif !icon_uri.to_s.include? "svg"
       image_tag icon_uri.to_s, class: "app-icon", title: icon_uri.to_s, "aria-hidden": true
+    else
+      svg_tag(icon_uri.to_s, class: 'app-icon')
     end
   end
 
+  def svg_tag(path, options={})
+    file = File.read(path)
+    doc = Nokogiri::HTML::DocumentFragment.parse(file)
+    svg = doc.at_css 'svg'
```

Comment:
Should probably need a begin/rescue block here with some default. Also a test case to ensure the default loads when the file can't either be read or parsed (or both).